src_tests/unittests/osm_configurator/model/project/calculation/test_aggregation_phase.py

```diff
@@ -0,0 +1,129 @@
+        # move the files from data to it
+        for file_name in os.listdir(path_old_data):
+            shutil.copy2(os.path.join(path_old_data, file_name), path_to_new_proejct)
+    else:
+        shutil.copy2(path_old_data, path_to_new_proejct)
+
+
+class TestAggregationPhase:
+    def test_corrupted_data(self):
+        # Set up paths
+        project_path: Path = Path(os.path.join(TEST_DIR, "build/aggregation_phase/projectXY"))
+
+        # Set up result folder from last phase
+        # so delete folder from previous test runs and copy the data into it
+        _prepare_previous_phase_folder(os.path.join(os.path.join(project_path, "results/"),
+                                                    calculation_phase_enum.CalculationPhase.ATTRACTIVITY_PHASE
+                                                    .get_folder_name_for_results()),
+                                       os.path.join(TEST_DIR, "data/monaco_split_up_files_invalid/"),
+                                       True
+                                       )
+
+        # Set up configurator
+        config_manager: ConfigurationManager = _prepare_config(project_path, [aggregation_method_enum_i.AggregationMethod.SUM])
+
+        # Execute test
+        phase: AggregationPhase = aggregation_phase_i.AggregationPhase()
+        result1: CalculationState = phase.calculate(config_manager)
+
+        assert result1[0] == calculation_state_enum.CalculationState.ERROR_INVALID_OSM_DATA
+
+    def test_correct_data(self):
+        # Set up paths
+        project_path: Path = Path(os.path.join(TEST_DIR, "build/aggregation_phase/projectXY"))
+
+        # Set up result folder from last phase
+        # so delete folder from previous test runs and copy the data into it
+        _prepare_previous_phase_folder(os.path.join(os.path.join(project_path, "results/"),
+                                                    calculation_phase_enum.CalculationPhase.ATTRACTIVITY_PHASE
+                                                    .get_folder_name_for_results()),
+                                       os.path.join(TEST_DIR, "data/0_traffic_cell_aggregation_test.csv"),
+                                       False
+                                       )
+
+        # Set up configurator
+        config_manager: ConfigurationManager = _prepare_config(project_path,
+                                                               [aggregation_method_enum_i.AggregationMethod.SUM])
+
+        # Execute test
+        phase: AggregationPhase = aggregation_phase_i.AggregationPhase()
+        result1: (CalculationState, str) = phase.calculate(config_manager)
+
+        assert result1[0] == calculation_state_enum.CalculationState.RUNNING
+
+        # Test if files were created
+        assert len(
+            os.listdir(os.path.join(project_path, "results/" + calculation_phase_enum.CalculationPhase.AGGREGATION_PHASE
+                                    .get_folder_name_for_results()))) == 1
+
```

Comment:
add test with more aggregationMethods and test if the file is correct

Reply:
fixed